### ISSUE TITLE
lightning panic ( runtime error: invalid memory address or nil point dereference)

### ISSUE THREAD

**Issue by fubinzh** [pingcap/br#1213]
Please answer these questions before submitting your issue. Thanks!

1. What did you do?
If possible, provide a recipe for reproducing the error.

  1. Use tidb-lightning to import data to TiDB.  (4 lightning node, each node imports 2.5 TB data.)
  

2. What did you expect to see?

Lightning import should be successful.

3. What did you see instead?

Lightning import failed with below errors

![lightning_panic](https://user-images.githubusercontent.com/7403864/121990033-b4318280-cdcf-11eb-8e98-7bed65891be5.png)

4. What version of BR and TiDB/TiKV/PD are you using?

tidb-lightning:v5.1.0-20210608

5. Operation logs
   - Please upload `br.log` for BR if possible
   - Please upload `tidb-lightning.log` for TiDB-Lightning if possible
   - Please upload `tikv-importer.log` from TiKV-Importer if possible
   - Other interesting logs

Logs can be found here: http://minio.pingcap.net:9000/minio/nfs/10TB_Testing/table31/

6. Configuration of the cluster and the task
   - `tidb-lightning.toml` for TiDB-Lightning if possible
   - `tikv-importer.toml` for TiKV-Importer if possible
   - `topology.yml` if deployed by TiUP

7. Screenshot/exported-PDF of Grafana dashboard or metrics' graph in Prometheus if possible

**Comment by gozssky** [comment]
The root cause may be related to [`kv.CommonHandle.Next()`](https://github.com/pingcap/tidb/blob/b8f6ac0dd30bba4f23b317891313f2921c825b30/kv/key.go#L270). 
```go
// Next implements the Handle interface.
func (ch *CommonHandle) Next() Handle {
	return &CommonHandle{
		encoded:       Key(ch.encoded).PrefixNext(),
		colEndOffsets: ch.colEndOffsets,
	}
}
```
```go
func (k Key) PrefixNext() Key {
	buf := make([]byte, len(k))
	copy(buf, k)
	var i int
	for i = len(k) - 1; i >= 0; i-- {
		buf[i]++
		if buf[i] != 0 {
			break
		}
	}
	if i == -1 {
		copy(buf, k)
		buf = append(buf, 0)
	}
	return buf
}
```
The next handle of a `CommonHandle` is probably  not a valid handle, because the `Next` implementation just add one to the encoded data. 

Lightning first call `nextKey` in https://github.com/pingcap/br/blob/05beea0674d5f4100de3ae80cb81948029eb65a3/pkg/lightning/backend/local/local.go#L1483.
Lightning second call `nextKey` for the same key in https://github.com/pingcap/br/blob/05beea0674d5f4100de3ae80cb81948029eb65a3/pkg/lightning/backend/local/local.go#L1623.

**Comment by gozssky** [comment]
Fixed by #1261.
